test/test_subscription.py

```diff
@@ -321,3 +331,55 @@ def test_continuous_subscription(self):
                 self.retrieve(8889, fname, fileVersion=2, targetFile=tmp_path())
         finally:
             subscription_listener.close()
+
+    def test_https_subscription(self):
+        if os.environ.get("NGAS_TESTDB") is not None:
+            pytest.skip("Need two clusters for this test to run")
+        cert_file = self.cert_auth.create_server_certificate(
+            [u"localhost", six.u(getHostName())], [ipaddress.IPv4Address(u"127.0.0.1")]
+        ).combined_filename
+        print("CERT FILE IS {}".format(cert_file))
+        # We configure the second server to send notifications via socket
+        # to the listener we start later
+        print("SETTING UP SERVER")
+        cfg = (('NgamsCfg.ArchiveHandling[1].EventHandlerPlugIn[1].Name', 'test.test_subscription.SenderHandler'),)
+
+        self.prepExtSrv(port=8778, cert_file=cert_file, stderr_level=4)
+        self.prepExtSrv(port=8779, cert_file=cert_file, stderr_level=4)
+        print("SET UP SERVER")
+
+        # Initial archiving
+        self.qarchive(8778, 'src/SmallFile.fits', mimeType='application/octet-stream')
+
+        # Version 2 of the file should only exist after
+        # subscription transfer is successful.
+        self.retrieve_fail(8779, 'SmallFile.fits', fileVersion=2, targetFile=tmp_path())
+
+        # Create listener that should get information when files get archives
+        # in the second server (i.e., the one on the receiving end of the subscription)
+        subscription_listener = self.notification_listener()
+
+        # Create subscription
+        params = {'url': 'https://localhost:8779/QARCHIVE',
+                  'subscr_id': 'HERE-TO-THERE',
+                  'priority': 1,
+                  'start_date': '%sT00:00:00.000' % time.strftime("%Y-%m-%d"),
+                  'concurrent_threads': 1}
+
+        self.assertEqual(
+            requests.get(
+                "https://127.0.0.1:8778/SUBSCRIBE", params=params
+            ).status_code, 200
+        )
+
+        # Do not like sleeps but xfer should happen immediately.
+        #try:
+        #    archive_evt = subscription_listener.wait_for_file(5)
+        #finally:
+        #    subscription_listener.close()
+        #self.assertIsNotNone(archive_evt)
+
+        #self.assertEqual(2, archive_evt.file_version)
+        #self.assertEqual('SmallFile.fits', archive_evt.file_id)
+
+        self.retrieve(8779, 'SmallFile.fits', fileVersion=1, targetFile=tmp_path())
```

Comment:
A general comment: this test is clearly based on `test_basic_subscription`, which sets up a single NGAS cluster with two servers and creates a subscription between the two. While this doesn't look like a real-life scenario, it's sufficient to test the subscription features. Is there any aspect of this test that would make that impossible?

In any case, this is, in my understanding, the whole reason of why you added initial https support to the server. As I mentioned in my first comment yesterday I'd rather simplify this, and replace the second NGAS server by a dummy, consume-everything-you-are-given https server, which additionally would send back a notification to the unit test once the data arrives. That would remove the need to have an https-enabled NGAS server, and allow us to separate the client-side https support (this PR) from the server-side.